py/selenium/webdriver/remote/webelement.py

```diff
@@ -313,25 +317,15 @@ def send_keys(self, *value):
             # file_input.send_keys(os.path.abspath("path/to/profilepic.gif"))
 
         """
+        ignore_files = kwargs.get('ignore_files', False)
         # transfer file to another machine only if remote driver is used
         # the same behaviour as for java binding
-        if self.parent._is_remote:
+        if self.parent._is_remote and not ignore_files:
             local_file = self.parent.file_detector.is_local_file(*value)
```

Comment:
@lukeis So, I solved it for our project, but I was thinking - since it made a pretty awful code IMO (try/finally/if): https://github.com/RedHatQE/cfme_tests/pull/2690/files#diff-54c758ec2117f433d2c05581244ceb74L756

What would you say if I implemented this behaviour as a context manager so one would just do:

```python
with webdriver.file_detector(UselessFileDetector):
    # do something
```

If it did not have such file detector already, it would create it, set it, run the code in the block and then set it back. If it had the same kind of detector already, it would do nothing. This would abstract the behaviour from the try/finally block.